feature-libs/product-configurator/rulebased/root/events/configurator-logout-event.listener.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, OnDestroy } from '@angular/core';
+import { EventService, LogoutEvent } from '@spartacus/core';
+import { merge, Subscription } from 'rxjs';
+import { ConfiguratorExpertModeService } from '../../core/services/configurator-expert-mode.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ConfiguratorLogoutEventListener implements OnDestroy {
```

Comment:
After core team confirms that event listener can be lazily loaded, and following @schuppel 's suggestion:
Let's have the event listener in the core entry point in feature-libs/product-configurator/rulebased/core/event.
Please make sure to export it to the public API of the core endpoint.
Create a dedicated module for the event listener and import this module in RulebasedConfiguratorCoreModule.
Create new action that removes all product bound configurations, the corresponding effect here /rulebased/core/state/effects/configurator-basic.effect.ts